src/averageAges.js

```diff
@@ -37,7 +39,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const onlyWomen = people.filter(person => (
+    withChildren
+      ? people.some(children => children.mother === person.name)
+      : person.sex === 'f'));
+
+  return onlyWomen.reduce((total, person) => (
+    calculateAge(total, person)), 0) / onlyWomen.length;
```

Comment:
Also you can create separate function

```suggestion
  return onlyWomen
    .reduce((total, person) => calculateAge(total, person), 0)
    / onlyWomen.length;
```